packages/format-library/src/link/inline.js

```diff
@@ -212,7 +212,17 @@ function InlineLinkUI( {
 	//  This caches the last truthy value of the selection anchor reference.
 	// This ensures the Popover is positioned correctly on initial submission of the link.
 	const cachedRect = useCachedTruthy( popoverAnchor.getBoundingClientRect() );
-	popoverAnchor.getBoundingClientRect = () => cachedRect;
+
+	// If the link is not active (i.e. it is a new link) then we need to
+	// override the getBoundingClientRect method on the anchor element
+	// to return the cached value of the selection represented by the text
+	// that the user selected to be linked.
+	// If the link is active (i.e. it is an existing link) then we allow
+	// the default behaviour of the popover anchor to be used. This will get
+	// the anchor based on the `<a>` element in the rich text.
+	if ( ! isActive ) {
+		popoverAnchor.getBoundingClientRect = () => cachedRect;
+	}
```

Comment:
If `isActive` is true then there is an existing link and thus we should allow `popoverAnchor ` (derived from `useAnchor`) to use the default implementation which will seek the `<a>` in the DOM and anchor to that.

Otherwise if we're creating a link then there is no `<a>` until the link is submitted. Therefore we need to cache the anchor value of the original text selection that the user selected to be linked.

Previously we just cached the selection value always, which wasn't correct.

Reply:
This fixes it for existing links, but for new links, the popover will have the same scroll issue since its position is cached.